wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -247,7 +245,7 @@ public void delete(@ApiParam("The workspace id") @PathParam("id") String id) thr
                                                                                         ConflictException,
                                                                                         ForbiddenException {
         if (!workspaceManager.getSnapshot(id).isEmpty()) {
-            machineManager.removeSnapshots(getCurrentUserId(), id);
+            machineManager.removeSnapshots(EnvironmentContext.getCurrent().getSubject().getUserId(), id);
```

Comment:
https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/dao/SnapshotDao.java#L78-78
namespace should be equal to user name, no?